src/transformers/modeling_utils.py

```diff
@@ -1845,7 +1858,7 @@ def gradient_checkpointing_disable(self):
         activations".
         """
         if self.supports_gradient_checkpointing:
-            self.apply(partial(self._set_gradient_checkpointing, value=False))
+            self.apply(partial(self._set_gradient_checkpointing, gradient_checkpointing_func=None))
```

Comment:
WHen we disable gradient checkpointing, I think the `module.gradient_checkpointing` will still be `True`. 
Let's make `module.gradient_checkpointing` into a property to be sure we always check if the function is none or not WDYT?

Reply:
Property could go at the ModelMixin level ?